src/gui/categoryfiltermodel.cpp

```diff
@@ -228,7 +228,7 @@ QVariant CategoryFilterModel::data(const QModelIndex &index, int role) const
 
 Qt::ItemFlags CategoryFilterModel::flags(const QModelIndex &index) const
 {
-    if (!index.isValid()) return 0;
+    if (!index.isValid()) return nullptr;
```

Comment:
Why nullptr? Is Qt::ItemFlags of pointer type?

Reply:
warning: zero as null pointer constant

Reply:
What compiler produces this warning?

Reply:
maybe return `{}` or `Qt::NoItemFlags`.

Reply:
It's "Clang Tidy and Clazy" in Qt creator 4.7. Dunno if it's something new or it got enabled by default with this release.

Reply:
I've disabled "Clang code model" in QtCreator 4.7, because it gives a lot of absurd warnings and prevents me from working. Try watching "session.cpp", for example.